### PR TITLE
[RDY] clang/"null pointer dereference": ex_cmds.c

### PR DESCRIPTION
**Pull request by ngortheone**
[clang report](https://neovim.io/doc/reports/clang/report-cbc443.html#EndPath)

**Problem:**
Based on the structure of the code clang suggests `ga.ga_data` could be `NULL` while `ga.ga_len > 0`, which will lead to null pointer deref.  

**Suggested solution:**
Slightly change the condition so that the code that clang complains on (true branch) is never executed when `ga_data == NULL`

### CODE REVIEW

File: src/nvim/ex_cmds.c [hunk 1]
```diff
@@ -5324,13 +5324,11 @@ static void helptags_one(char_u *const dir, const char_u *const ext,
 
   FreeWild(filecount, files);
 
-  if (!got_int) {
+  if (!got_int && ga.ga_data != NULL) {
     /*
      * Sort the tags.
      */
-    if (ga.ga_data != NULL) {
```
Comment (ngortheone) [inline src/nvim/ex_cmds.c]: This suggests that execution of the function continues when `ga_data ==NULL`

Line 5339: for clang here is potential null pointer dereference, as connection between `ga_len` and `ga_data` is implicit

**Comment by justinmk** [comment]
LGTM. lint?

**Comment by ngortheone** [comment]
ack

**Comment by ngortheone** [comment]
linter still barks on this line
`fprintf(fd_tags, "%s\t/*", s);` 
complaining on comment style..

File: src/nvim/ex_cmds.c [hunk 2]
```diff
@@ -5357,31 +5345,31 @@ static void helptags_one(char_u *const dir, const char_u *const ext,
       fprintf(fd_tags, "!_TAG_FILE_ENCODING\tutf-8\t//\n");
     }
```
Comment (justinmk) [inline src/nvim/ex_cmds.c]: In the case of "no tags", before this change, this line would still write to the  mostly-empty file. Now it doesn't. That's probably ok I guess.

**Pull request merged**